web/ajax/stream.php

```diff
@@ -47,12 +47,16 @@
 
 $remSockFile = ZM_PATH_SOCKS.'/zms-'.sprintf("%06d",$_REQUEST['connkey']).'s.sock';
 $max_socket_tries = 10;
-while ( !file_exists($remSockFile) && $max_socket_tries-- ) //sometimes we are too fast for our own good, if it hasn't been setup yet give it a second.
-	usleep(200000);
+while ( !file_exists($remSockFile) && $max_socket_tries-- ) { //sometimes we are too fast for our own good, if it hasn't been setup yet give it a second.
+  usleep(200000);
+}
 
-if ( !@socket_sendto( $socket, $msg, strlen($msg), 0, $remSockFile ) )
-{
+if ( !file_exists($remSockFile) ) {
+  ajaxError("Socket $ramSocketFile does not exist.  This file is created by zms, and since it does not exist, either zms did not run, or zms exited early.  Please check your zms logs and ensure that CGI is enabled in apache and check that the PATH_ZMS is set correctly.  Make sure that ZM is actually recording.  If you are trying to view a live stream and the capture process (zmc) is not running then zms will exit. Please go to http://zoneminder.readthedocs.io/en/latest/faq.html#why-can-t-i-see-streamed-images-when-i-can-see-stills-in-the-zone-window-etc for more information.");
```

Comment:
Whew, that's a long message. Out of curiosity, have you simulated this error condition to see how well this message shows up in the log window?